### ISSUE TITLE
Jenkins-test-harness won't find jenkins war with Gradle 7.6

### ISSUE THREAD

**Issue by jiridanek** [jenkinsci/jenkins-test-harness#551]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```text
n/a
  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Java

### Reproduction steps

The jenkins test harness uses the following to locate the jenkins war file

https://github.com/jenkinsci/jenkins-test-harness/blob/6d05a3953e0d59b4a318d20efed8c4bfaee875c3/src/main/java/org/jvnet/hudson/test/WarExploder.java#L80-L91

Gradle is using hashes in the local jar cache, https://discuss.gradle.org/t/how-are-hash-values-for-gradle-repositories-calculated/4345

So, this does not work with Gradle 7.6, because it assumes a directory hierarchy of a Maven local repository, but that version of Gradle has also the filehash in the path, so I get correct value for `core`, but wrong value for `version`:

```
core = /home/jdanek/.gradle/caches/modules-2/files-2.1/org.jenkins-ci.main/jenkins-core/2.361.2/46513df8e2016ef981a2608c9e362a5b3eab112e/jenkins-core-2.361.2.jar
version = 46513df8e2016ef981a2608c9e362a5b3eab112e
```

and now, `core.getName().equals("jenkins-core-" + version + ".jar") && core.getParentFile().getParentFile().getName().equals("jenkins-core")` has no chance of working, because first, version is wrong, and the path also misses one extra `getParentFile()`.

### Expected Results

Locate the war successfully even with Gradle 7.6.

### Actual Results

```
Test timeout disabled.
=== Starting loadJenkinsCasCTest(io.strimzi.ci.integration.casc.JCasCTest)

/home/jdanek/.gradle/caches/modules-2/files-2.1/org.jenkins-ci.main/jenkins-core/2.361.2/46513df8e2016ef981a2608c9e362a5b3eab112e/jenkins-core-2.361.2.jar is not in the expected location, and jenkins-war-*.war was not in /home/jdanek/.gradle/caches/7.6/workerMain/gradle-worker.jar:/home/jdanek/repos/testing/ci/msg-qe-cci/jobs/dsl/amq-clients/build/classes/groovy/test:/home/jdanek/repos/testing/ci/msg-qe-cci/jobs/dsl/amq-clients/build/resources/test:/home/jdanek/repos/testing/ci/msg-qe-cci/jobs/dsl/amq-clients/build/classes/groovy/main:/home/jdanek/repos/testing/ci/msg-qe-cci/jobs/dsl/amq-clients/build/resources/main:/home/jdanek/.gradle/caches/modules-2/files-2.1/org.jenkins-ci.main/jenkins-war/2.361.2/9b06767a2ca1cd96aa9686c766e4a83a2254e857/jenkins-war-2.361.2.war:/home/jdanek/.gradle/caches/modules-2/files-2.1/org.jenkins-ci.main/jenkins-core/2.361.2/46513df8e2016ef981a2608c9e362a5b3eab112e/jenkins-core-2.361.2.jar:/home/jdanek/.gradle/caches/modules-2/files-2.1/org.jenkins-ci.plugins/job-dsl/1.81/5e62a0d531e92e67dfa47058b752909f1375396c/job-dsl-1.81.jar:/home/jdanek/.gradle/caches/modules-2/files-2.1/org.jenkins-ci.plugins/job-dsl-core/1.81/889444d2ff6f84418a1c21c5a31cb2a5786b1ab8/job-dsl-core-1.81.jar:/home/jdanek/.gradle/caches/modules-2/files-2.1/com.lesfurets/jenkins-pipeline-unit/1.14/a39ac387827541a0aaafc1a416cee48a049efe10/jenkins-pipeline-unit-1.14.jar:/home/jdanek/.gradle/caches/modules-2/files-2.1/org.jenkins-ci.plugins/ghprb/1.42.2/c5771ab8986b02811a9e3fa587e20b6c99068eb1/ghprb-1.42.2.hpi:/home/jdanek/.gradle/caches/modules-2/files-2.1/org.jenkins-

[...]

2.1/org.mozilla/rhino/1.7.7.2/a7c4a9ba8b6922374580d71060ef71eafa994256/rhino-1.7.7.2.jar:/home/jdanek/.gradle/caches/modules-2/files-2.1/javax.xml.stream/stax-api/1.0-2/d6337b0de8b25e53e81b922352fbea9f9f57ba0b/stax-api-1.0-2.jar:/home/jdanek/.gradle/caches/modules-2/files-2.1/pull-parser/pull-parser/2/c2f040085bfcd35963e9504bec11dac57cd0a86e/pull-parser-2.jar:/home/jdanek/.gradle/caches/modules-2/files-2.1/com.ibm.icu/icu4j/59.1/6f06e820cf4c8968bbbaae66ae0b33f6a256b57f/icu4j-59.1.jar:/home/jdanek/.gradle/caches/modules-2/files-2.1/org.apache.geronimo.specs/geronimo-jms_1.1_spec/1.1.1/c872b46c601d8dc03633288b81269f9e42762cea/geronimo-jms_1.1_spec-1.1.1.jar:/home/jdanek/.gradle/caches/modules-2/files-2.1/org.apache.geronimo.specs/geronimo-j2ee-management_1.1_spec/1.0.1/5372615b0c04c1913c95c34a0414cef720ca2855/geronimo-j2ee-management_1.1_spec-1.0.1.jar:/home/jdanek/.gradle/caches/modules-2/files-2.1/javax.activation/javax.activation-api/1.2.0/85262acf3ca9816f9537ca47d5adeabaead7cb16/javax.activation-api-1.2.0.jar
	at org.jvnet.hudson.test.WarExploder.findJenkinsWar(WarExploder.java:90)
	at org.jvnet.hudson.test.WarExploder.explode(WarExploder.java:116)
	at org.jvnet.hudson.test.WarExploder.getExplodedDir(WarExploder.java:56)
	at org.jvnet.hudson.test.JenkinsRule._createWebServer(JenkinsRule.java:803)
	at org.jvnet.hudson.test.JenkinsRule.createWebServer(JenkinsRule.java:758)
	at org.jvnet.hudson.test.JenkinsRule.createWebServer(JenkinsRule.java:747)
	at org.jvnet.hudson.test.JenkinsRule.newHudson(JenkinsRule.java:699)
	at org.jvnet.hudson.test.JenkinsRule.before(JenkinsRule.java:415)
	at io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule.before(JenkinsConfiguredWithCodeRule.java:23)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:608)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:62)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:113)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:65)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
```

### Anything else?

I implemented the following workaround in my `build.gradle.kts`

```kotlin
tasks.named<Test>("test") {
    // set jenkins-war path as system property
    val artifact = testWar.resolvedConfiguration.resolvedArtifacts.find {
        it.moduleVersion.id.group == "org.jenkins-ci.main"
                && it.moduleVersion.id.module.name == "jenkins-war"
    }
    systemProperty("jth.jenkins-war.path", artifact!!.file.absolutePath)

    // ...
}
```

**Comment by basil** [comment]
Fixed in #561. Released in [1947.vd2a_0b_38736ea_](https://github.com/jenkinsci/jenkins-test-harness/releases/tag/1947.vd2a_0b_38736ea_).

---

You included no steps to reproduce the problem from scratch with a minimal reproducible example (MRE), which made dealing with this issue much more difficult than it needed to be. As a one-time courtesy, I have resolved the issue despite the lack of an MRE. In the future, include complete steps to reproduce the problem from scratch with an MRE when filing issues.

**Comment by jiridanek** [comment]
> Fixed in #561. Released in [1947.vd2a_0b_38736ea_](https://github.com/jenkinsci/jenkins-test-harness/releases/tag/1947.vd2a_0b_38736ea_).
> 
> You included no steps to reproduce the problem from scratch with a minimal reproducible example (MRE), which made dealing with this issue much more difficult than it needed to be. As a one-time courtesy, I have resolved the issue despite the lack of an MRE.

You could've just asked for it...

"The problem wasn't that someone silently helped out, but that there was no indication of what was going on, and that the help wouldn't be there every time." https://prog21.dadgum.com/40.html

> In the future, include complete steps to reproduce the problem from scratch with an MRE when filing issues.

Well, poor @jimklimov wrote a fix in https://github.com/jenkinsci/jenkins-test-harness/pull/423 and that effort largely went to waste...

What usually works for me is to write a MRID (Minimal Reasonable Issue Description) and then, if it catches interest, provide the needed details. It is sad to see your MRE bitrot over the months nobody works on the issue. Or see it not used because the problem was "pretty obvious from the stack trace alone".
